File: python/space-age/space_age.py
from functools import partial

EARTH_YEAR_IN_SECONDS = 31557600
MULTIPLIERS = {
    'on_mercury': 0.2408467,
    'on_venus': 0.61519726,
    'on_earth': 1.0,
    'on_mars': 1.8808158,
    'on_jupiter': 11.862615,
    'on_saturn': 29.447498,
    'on_uranus': 84.016846,
    'on_neptune': 164.79132
}

class SpaceAge(object):
    def __init__(self, age):
        self.seconds = age
        for planet in MULTIPLIERS:
            setattr(self, planet, partial(self.__age, planet))

    @staticmethod
    def __period(planet):
        return EARTH_YEAR_IN_SECONDS * MULTIPLIERS[planet]

    def __age(self, planet):
        age = self.seconds / self.__period(planet)
        return round(age, 2)

File: csharp/space-age/SpaceAge.cs
using System;

public class SpaceAge
{
    private const double EARTH_YEAR_IN_SEC = 31557600;
    private double age;

    public SpaceAge(long seconds)
    {
        age = (double)seconds;
    }

    public double OnEarth() =>
        AgeFromPeriod();

    public double OnMercury() =>
        AgeFromPeriod((double)0.2408467);

    public double OnVenus() =>
        AgeFromPeriod((double)0.61519726);

    public double OnMars() =>
        AgeFromPeriod((double)1.8808158);

    public double OnJupiter() =>
        AgeFromPeriod((double)11.862615);

    public double OnSaturn() =>
        AgeFromPeriod((double)29.447498);

    public double OnUranus() =>
        AgeFromPeriod((double)84.016846);

    public double OnNeptune() =>
        AgeFromPeriod((double)164.79132);

    private double AgeFromPeriod(double period = 1) =>
        age / EARTH_YEAR_IN_SEC / period;
}